### ISSUE TITLE
Resets the top menu settings when updating

### ISSUE THREAD

**Issue by ssl1984** [modxcms/revolution#15327]
After updating the system, the top menu settings are reset. If the system items were moved to another partition or root, they are lost.

**Comment by Ruslan-Aleev** [comment]
Yes, it is also annoying that the menu items change their positions.
By the way, the same happens after updating a package.

**Comment by Ibochkarev** [comment]
Still happens on a core upgrade. @Ruslan-Aleev is right that positions move, and that a package upgrade can do the same thing to the items that package ships.

Core menus are built in `_build/data/transport.core.menus.php`. `buildMenus()` sets `parent` and `menuindex` from that tree. `_build/transport.core.php` packs them with `UPDATE_OBJECT => true` and unique key `text`, including the `Children` relation. On install, `xPDOObjectVehicle::_installObject()` finds the existing row and `fromArray()` copies the packaged fields over it. For a child it also sets `parent` from the packaged parent before that copy. A system item that was moved under another parent, or reordered, is written back to the tree in the package. `action`, `handler`, `permissions`, `icon`, and `description` are updated by that same copy, which is why a plain `UPDATE_OBJECT => false` would also skip real menu fixes. Intentional moves already have their own upgrade scripts, such as `setup/includes/upgrades/common/3.1.0-trash-menu.php`, which inserts `trash` only when that row is missing.

Before `fromArray()`, the vehicle stores the current row on `$transport->_preserved`. Add a resolve script on the core menu vehicle that, for each preserved row with a `text` key, writes `parent` and `menuindex` back from that copy and saves. New items have no preserved row, so they keep the parent and index from the package.

The same vehicle path is what a package update uses. Changing xPDO for every package would also block an extra that really does move its own item during an upgrade. An extra that wants to keep a custom parent and index has to preserve those two fields in its own menu vehicle.
